lib/staging/tls/tls.hpp

```diff
@@ -221,6 +222,19 @@ class Connection {
      * \note the certificate must not be freed
      */
     [[nodiscard]] const Certificate* peer_certificate() const;
+
+    /**
+     * \brief obtain the underlying SSL context
+     * \returns the underlying SSL context pointer
+     */
+    [[nodiscard]] SSL* ssl_context() const;
+
+    /**
+     * \brief set the read timeout in ms
+     */
+    void set_read_timeout(int ms) {
```

Comment:
Not sure this is needed.  
`[[nodiscard]] result_t read(std::byte* buf, std::size_t num, std::size_t& readbytes, int timeout_ms);`
supports a different timeout ...
Full non-blocking IO is supported, so may not be needed at all.